api/envoy/type/matcher/metadata.proto

```diff
@@ -68,8 +68,6 @@ import "validate/validate.proto";
 // <envoy_api_msg_config.rbac.v2alpha.Principal>`.
 message MetadataMatcher {
   // Specifies the segment in a path to retrieve value from Metadata.
-  // Note: Currently it's not supported to retrieve a value from a list in Metadata. This means it
```

Comment:
I think this comment should remain, we still don't support to retrieve a value from a list. 

And the comment in line 80 should be updated that it now supports to compare a value to a list, but such comparison only supports primitive values in a list, embedded values in a list are still not supported.

Reply:
@liminw can we revert the comment?

Reply:
I modified the origin comment to make it more accurate.